### ISSUE TITLE
Enhancement to make --trail take an optional argument

### ISSUE THREAD

**Issue by andrewjmaguire** [rjbs/Dist-Zilla#640]
It would be nice if the dzil build & release --trial option could accept an optional integer argument that would be appended after -TRIAL in the filename.

**Comment by karenetheridge** [comment]
To what end? Just to make the tarball filename unique? Wouldn't it already be unique given the presence of a version number?

**Comment by andrewjmaguire** [comment]
No, the filename is not unique if you just use a two digit version number x.y in your dist.ini file.
As I understand, PAUSE states that development versions can be uploaded if you "choose a filename that matches /\d\.\d+_\d/ or matches -TRIAL[0-9]* before the file extension".
So an optional number is acceptable after the -TRIAL.
I note that if you use x.y_z then dzil build --trial does not do anything different to dzil build - there is already a 3rd digit to identify development versions.
However, if you just use x.y then dzil build and dzil build--trial will create two different tar files.
But once you upload the file to PAUSE, using dzil release --trial, you cannot upload a subsequent build because the file name will have stayed the same.
So instead of being able to use dzil release --trial again for a new version, I ran dzil build --trial and renamed the generated tar ball to MyPackage-1.0-TRIAL1.tar.gz and manually uploaded that to PAUSE.
